### PR TITLE
Don't require to pass a registry on MetricsFactory initialization

### PR DESCRIPTION
**Pull request by fabianfett**
Do not require users to explicitly pass a `PrometheusCollectorRegistry` on `PrometheusMetricsFactory` creation.

### CODE REVIEW

File: Sources/Prometheus/PrometheusMetricsFactory.swift [hunk 1]
```diff
@@ -16,8 +16,16 @@ import CoreMetrics
...
+    private static let _defaultRegistry = PrometheusCollectorRegistry()
+
+    /// The default ``PrometheusCollectorRegistry``, which is used inside the ``PrometheusMetricsFactory``
+    /// if no other is provided in ``init(client:)`` or set via ``PrometheusMetricsFactory/client``
+    public static var defaultRegistry: PrometheusCollectorRegistry {
+        self._defaultRegistry
+    }
+
+    /// The underlying ``PrometheusCollectorRegistry`` that is used to generate the swift-metrics handlers
+    public var registry: PrometheusCollectorRegistry
```
Comment (FranzBusch) [inline Sources/Prometheus/PrometheusMetricsFactory.swift:28]: Do we want this to be public?

Comment (fabianfett) [reply Sources/Prometheus/PrometheusMetricsFactory.swift:28]: Yes. It should! So that users can access what they got by default.

Comment (ktoso) [reply Sources/Prometheus/PrometheusMetricsFactory.swift:28]: I don't mind I guess :)

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
conflict but LGTM

**Pull request merged**